### PR TITLE
Added rspec to the project

### PR DESCRIPTION
**Pull request by seanstory**
Closes https://github.com/elastic/enterprise-search-team/issues/1305

Adds the ability to run rspec tests along with jvm tests through maven

### CODE REVIEW

File: .ci/script/build.sh [hunk 1]
```diff
@@ -28,6 +28,16 @@ rbenv versions
 rbenv global $JRUBY_VERSION
 ruby --version # just to double-check...
 which -a ruby
+ruby_bin_file=$(which ruby)
+export BUNDLE_CMD="$ruby_bin_file -S bundle"
```
Comment (seanstory) [inline .ci/script/build.sh:32]: @tarekziade is there a better way to invoke this logic from https://github.com/elastic/ent-search-ci-build/blob/main/lib/entsearchci/generator/ruby.rb without having to use `perform-ci-build`?

File: Gemfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
+
+# Pull gem index from rubygems
+source 'https://rubygems.org'
+
+# Pin the version of bundle we support
+gem 'bundler', '2.2.29'
+
+group :test do
+  gem 'rspec-core', '~> 3.10.1'
+  gem 'rspec-rails', '~> 4.0.2'
```
Comment (orhantoy) [inline Gemfile:10]: Do we need this gem or could RSpec core suffice?

Comment (seanstory) [reply Gemfile]: good calll, we don't need `rspec-rails`. The `rspec-collections-matchers` I leave though, as that gives us the `expect()` function, and the `rspec_junit_formatter` should allow jenkins to give us a pretty breakdown of our individual test outputs.

**Comment by seanstory** [comment]
Woo! Rspec tests running in CI, and their output is aggregated with other Junit/maven test output!

Screen Shot 2022-02-08 at 3 41 16 PM

File: hello-world-connector/src/test/ruby/example_test.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+RSpec.describe 'example' do
+    it "can run a simple test" do
+      expect(1).to eq(1)
+    end
```
Comment (orhantoy) [inline hello-world-connector/src/test/ruby/example_test.rb:4]: Indentation is first 4 and then 2 at the next level; both should probably be 2.

File: hello-world-connector/src/test/ruby/example_test.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+end
```
Comment (orhantoy) [inline hello-world-connector/src/test/ruby/example_test.rb:5]: Final newline is also missing here.

File: Gemfile [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+source 'https://rubygems.org'
+
+# Pin the version of bundle we support
+gem 'bundler', '2.2.29'
+
+group :test do
+  gem 'rspec-core', '~> 3.10.1'
+  gem 'rspec-collection_matchers', '~> 1.2.0'
+  gem 'rspec_junit_formatter'
+end
```
Comment (orhantoy) [inline Gemfile:12]: Final newline is also missing here. VSCode and probably other editors insert this by default.

File: Gemfile [hunk 6]
```diff
@@ -0,0 +1,12 @@
+
```
Comment (orhantoy) [inline Gemfile:1]: Let's remove this empty line.

**Review by orhantoy — COMMENTED** [review]
Looks good; just found some minor stylistic things.

**Review by orhantoy — APPROVED** [review]
🚀

**Pull request merged**